distributed/comm/ucx.py

```diff
@@ -23,16 +24,24 @@
 
 logger = logging.getLogger(__name__)
 
-
 # In order to avoid double init when forking/spawning new processes (multiprocess),
 # we make sure only to import and initialize UCX once at first use. This is also
 # required to ensure Dask configuration gets propagated to UCX, which needs
 # variables to be set before being imported.
-ucp = None
+if TYPE_CHECKING:
+    try:
+        import ucp
+        from ucp import create_endpoint as ucx_create_endpoint
+        from ucp import create_listener as ucx_create_listener
+    except ImportError:
+        pass
+else:
+    ucp = None  # type: ignore
+    ucx_create_endpoint = None  # type: ignore
+    ucx_create_listener = None  # type: ignore
+
```

Comment:
@pentschev @madsbk @quasiben could one of you review the changes to this module?

Reply:
> However, it seems it's only being imported here when TYPE_CHECKING is defined, when is that going to happen in Distributed?

Today we don't run any type checking as part of the development process, but after https://github.com/dask/distributed/pull/5348 is merged we'll start running `mypy` as a `pre-commit` hook. Given that this `ucp` import will only happen during type checking, do you think the proposed changes here are okay?

Reply:
If it's guaranteed that `TYPE_CHECKING is False` for all runtime situations, then I think this is no problem.

Reply:
That should be the case. From https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING:

> It is False at runtime

Reply:
In that case, we should be good for now. If there's any reason why `TYPE_CHECKING is True` at runtime, we'll need to make changes to cover that.